.github/workflows/ci.yml

```diff
@@ -262,19 +262,13 @@ jobs:
         run: echo ${{ matrix.env }} >> $GITHUB_ENV
 
       # Build the tests before running to improve cross compilation speed
-      - name: Run cargo build
-        uses: actions-rs/cargo@v1.0.3
-        with:
-          command: build
-          args: --tests ${{ matrix.exclude }} ${{ matrix.target != 'native' && format('--target {0}', matrix.target) || '' }} ${{ matrix.args }}
-          use-cross: ${{ matrix.target != 'native' }}
+      - name: Run cargo/cross build
+        run: |
+          ${{ matrix.target != 'native' && 'cross' || 'cargo' }} build --tests ${{ matrix.exclude }} ${{ matrix.target != 'native' && format('--target {0}', matrix.target) || '' }} ${{ matrix.args }}
 
-      - name: Run cargo test
-        uses: actions-rs/cargo@v1.0.3
-        with:
-          command: test
-          args: ${{ matrix.exclude }} ${{ matrix.target != 'native' && format('--target {0}', matrix.target) || '' }} ${{ matrix.args }}
-          use-cross: ${{ matrix.target != 'native' }}
+      - name: Run cargo/cross test
+        run: |
+          ${{ matrix.target != 'native' && 'cross' || 'cargo' }} test ${{ matrix.exclude }} ${{ matrix.target != 'native' && format('--target {0}', matrix.target) || '' }} ${{ matrix.args }}
```

Comment:
This is a bit verbose. I could split this into different tasks but then we would be repeating ourselves, which given the long list of args seems worse.

```
- name: build with cargo
  if: ${{ matrix.target == 'native' }}
  run: cargo build

- name: build with cross
  if: ${{ matrix.target != 'native' }}
  run: cross build

```